user-guide/Advanced_Functionality/Failover/Configuring_Failover/Preparing_the_two_DataMiner_Agents.md

```diff
@@ -50,3 +52,78 @@ Addresses: 10.11.5.52
+
+   1. Copy the contents of the `DataBases` tag.
+
+   1. Paste this content in the file `C:\Skyline DataMiner\db.xml` on the backup DMA.
+
+      > [!IMPORTANT]
+      > If multiple [OpenSearch clusters](xref:Configuring_multiple_OpenSearch_clusters) or [Elasticsearch clusters](xref:Configuring_multiple_Elasticsearch_clusters) are used prior to DataMiner 10.4.0/10.4.2, there is an additional file that needs to be copied from the primary to the backup DMA. You can find this *DBConfiguration.xml* file in the folder `C:\Skyline DataMiner\Databases\`. Make sure you also do this while the backup Agent is stopped.
+
+      > [!NOTE]
+      > If multiple [OpenSearch clusters](xref:Configuring_multiple_OpenSearch_clusters) or [Elasticsearch clusters](xref:Configuring_multiple_Elasticsearch_clusters) are used, you may wish to configure the *priorityOrder* attribute differently on the main or backup DMA. You can do this if you want to change which indexing cluster is read from when there is a Failover switch. For more info on the *priorityOrder* attribute, see [OpenSearch clusters](xref:Configuring_multiple_OpenSearch_clusters) or [Elasticsearch clusters](xref:Configuring_multiple_Elasticsearch_clusters).
+
+   1. Restart the backup DMA.
+
+1. If TLS is used, make sure that the required certificates are imported, and this is correctly configured on the backup DMA as well.
+
+   - For Cassandra see [Encryption in Cassandra](xref:Security_Cassandra_TLS).
+
+   - For OpenSearch, see [Securing the OpenSearch database](xref:Security_OpenSearch).
+
+   - For Elasticsearch (deprecated) see [Securing the Elasticsearch database](xref:Security_Elasticsearch).
+
+### Separate Cassandra setup with indexing
+
+We **recommend against this setup** for Failover. With this setup, it is possible that if one node goes down, all data in the indexing database is unavailable.
+
+If you do go ahead with this, take into account that the the indexing database will automatically create a cluster of two nodes: one for the primary DMA and one for the backup DMA. To ensure this can happen, make sure that **ports 9200 and 9300** are open between primary and backup DMA so the databases can communicate. See [Configuring the IP network ports](xref:Configuring_the_IP_network_ports).
+
+For Cassandra, you will need to take the same steps as detailed below for a [separate Cassandra setup without indexing](xref:Preparing_the_two_DataMiner_Agents#separate-cassandra-setup-without-indexing).
+
+### Separate Cassandra setup without indexing
+
+When you set up Failover, DataMiner will automatically attempt to add Cassandra to a cluster of two Cassandra nodes, one for the primary DMA and one for the backup DMA. To make sure this can happen correctly, follow the steps below before you attempt to set up Failover:
+
+1. Make sure that Cassandra is installed and running on both DataMiner Agents, by opening *Windows Services* and checking whether the *cassandra* service exists and is running.
```

Comment:
What if it's not installed? Can we refer the user to https://docs.dataminer.services/user-guide/Advanced_Functionality/Databases/Configuring_dedicated_clustered_storage/Migrating_existing_systems/Migrating_the_general_database_to_Cassandra.html to resolve this?

Reply:
I don't believe so. I think the most similar procedure is the one found in https://docs.dataminer.services/user-guide/Advanced_Functionality/Security/Advanced_security_configuration/Database_security/Security_Cassandra_general/Cassandra_updating.html?q=updating+cassandra&tabs=tabid-1 in the "On Windows" tab.

Reply:
That page explains how to update Cassandra, but not how to install it, and this step refers to checking whether it is installed. I assume installation either happens when you install DataMiner, or if you used a legacy installer, when you migrate. Hence my suggestion to refer to the migration page.

Reply:
Indeed, using an old installer, it would be possible that you'd still have to migrate. In that case, indeed the migration would work. The page I mentioned does also install Cassandra using the script _cd 'C:\Program Files\Cassandra\bin\'; .\cassandra.ps1 -install_ and as such, an adapted version of it could have been used under certain circumstances.

Reply:
I've added a note with this info. Does this look OK to you?

Reply:
Looks ok!